x-pack/plugins/observability_solution/metrics_data_access/common/inventory_models/types.ts

```diff
@@ -235,29 +235,34 @@ export type TSVBMetricModelCreator = (
   interval: string
 ) => TSVBMetricModel;
 
-export const ESBasicMetricAggRT = rt.record(
-  rt.string,
-  rt.union([
-    rt.undefined,
-    rt.type({
-      field: rt.string,
-    }),
-  ])
-);
-
-export const ESDerivativeAggRT = rt.type({
-  derivative: rt.type({
-    buckets_path: rt.string,
-    gap_policy: rt.keyof({ skip: null, insert_zeros: null }),
-    unit: rt.string,
-  }),
-});
+export const isBasicMetricAgg = (
+  agg: unknown
+): agg is Record<string, undefined | Pick<estypes.AggregationsMetricAggregationBase, 'field'>> => {
+  if (agg && typeof agg === 'object' && Object.keys(agg).length !== 0) {
+    for (const key in agg) {
+      if (Object.hasOwn(agg, key)) {
+        return !!(
+          agg[key as keyof typeof agg] === undefined ||
+          (agg[key as keyof typeof agg] as estypes.AggregationsMetricAggregationBase).field
+        );
+      }
+    }
+  }
+
+  return false;
```

Comment:
Nit: we can reverse the order and make this as early exit in case the condition is not met - wdyt?

Reply:
I did it initially, but TS complained about missing return statement so I had to put `return false` at the end anyway. 

Error: 
<img width="867" alt="image" src="https://github.com/user-attachments/assets/0ae52f5d-5675-4a2c-9e9e-a85ec18e2992">

so it ended up like this:
<img width="806" alt="image" src="https://github.com/user-attachments/assets/cfc0ed05-3c7b-41b5-bacf-d6e9c0192339">

Which I didn't like. I found it less readable. Which apprach do you find better?

Reply:
The last change 
<img width="713" alt="image" src="https://github.com/user-attachments/assets/3667b0f8-5d86-44a5-9137-3aded06a1c6f">

looks good! Thank you for improving it 💯